File: Lab1/lab1.java
//Kevin Farokhrouz
//1002072886
//Java 1.8.0_412 (Object Oriented language)
//Win 11

import java.io.File;

public class kaf2886_lab01{
	
	//Returns the file size in bytes
	public static long getSize(File file) {
		return file.length();
	}
	
	public static long direcSize(File folder) {
		long sum = 0;
		//Lists the contents of the folder. This includes subfolders and files.
		String contents[] = folder.list();

		//Error Handling
		if (contents == null) {
			return 0;
		}
		
		for(int i = 0; i < contents.length; i++) {
			// Skip "." and ".."
			if (contents[i].equals(".") || contents[i].equals("..")) {
				continue;
			}
			//The entry path is constructed using the parent folder + entryName
			File entry = new File(folder + "/" + contents[i]);
			//Recurse through the entry if it is a subfolder
			if(entry.isDirectory()){
				sum = sum + direcSize(entry);		
			}
			else {
				//Else (the entry is a file), increment sum by the file size in bytes.
				sum = sum + getSize(entry);
			}	
		}
		return sum;
	}

	public static void main(String args[]) {
		//Functions take type File as input, format the path as a File
		File cwd = new File(".");
		//Int might not be enough for large folders. If your folder size doesn't fit inside of a long, god help you.
		long sum = direcSize(cwd);
		System.out.println(sum);	
	}
}

File: Lab1/lab1.py
#Kevin Farokhrouz
#1002072886
#Python 3.11.9 (Scripting language)
#Win 11

import os


#Uses os.path.getsize to return the size of the file at filePath
def getSize(filePath):
    return os.path.getsize(filePath) 
    
#Computes the total size, in bytes, of all of the files located in all subfolders beginning at folder
def dirSpace(folder):
    sum = 0
    #Creates a list of every entry inside of folder
    entries = os.listdir(folder)
    for entry in entries:
        #Skip . and ..
        if entry == "." or entry == "..":
            continue
        #The path to the entry is the current path + " /" + the entry name
        entry = os.path.join(folder, entry)
        #If the entry is a file, increment sum by the file's size
        if os.path.isfile(entry):
            sum += getSize(entry)
        #If the entry is a folder, recurse on the folder and increment sum by the size of all files in all subfolders beginning at entrypath
        else:
            sum += dirSpace(entry)
    return sum

#Begin recursion at cwd
sum = dirSpace(".")
print(sum)